OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -574,9 +572,156 @@ public void TestGetVariationForFeatureWhenTheUserIsNotBucketedIntoFeatureExperim
 
             Assert.AreEqual(expectedVariation, actualVariation);
 
-            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" was bucketed into a rollout for feature flag \"string_single_variable_feature\"."));
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is bucketed into a rollout for feature flag \"string_single_variable_feature\"."));
+        }
+        
+        [Test]
+        public void TestGetVariationForFeatureWhenTheUserIsNeitherBucketedIntoFeatureExperimentNorToFeatureRollout()
+        {
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("string_single_variable_feature");
+
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureExperiment(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureRollout(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeature(featureFlag, "user1", new UserAttributes());
+            Assert.IsNull(actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is not bucketed into a rollout for feature flag \"string_single_variable_feature\"."));
+        }
+
+        [Test]
+        public void TestGetVariationForFeatureRolloutWhenRolloutIsNotInDataFile()
+        {
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("boolean_feature");
+            var invalidRolloutFeature = new FeatureFlag {
+                RolloutId = "invalid_rollout_id",
+                Id = featureFlag.Id,
+                Key = featureFlag.Key,
+                ExperimentIds = new List<string>(featureFlag.ExperimentIds),
+                Variables = featureFlag.Variables
+                };
+
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureExperiment(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeatureRollout(featureFlag, "user1", new UserAttributes());
+            Assert.IsNull(actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The feature flag \"boolean_feature\" is not used in a rollout."));
+        }
+
+        [Test]
+        public void TestGetVariationForFeatureRolloutWhenRolloutDoesNotHaveExperiment()
+        {
```

Comment:
Please assert the message. that no experiment in the rollout.

Reply:
No message has been logged in this case.